test/e2e/wmco_test.go

```diff
@@ -131,3 +146,68 @@ func testWindowsNodeDeletion(t *testing.T) {
 	}
 	defer testCtx.Cleanup()
 }
+
+// testWMCValidation tests if validations of the fields of WindowsMachineConfigs CRD are working as expected
+func testWMCValidation(t *testing.T) {
+	var wmcReplicasFieldValidationTests = []struct {
+		name                       string
+		isReplicasFieldPresent     bool
+		replicasFieldValue         int
+		isTestExpectedToThrowError bool
+		expectedErrorInTest        string
+	}{
+		// We are only checking negative test cases here, positive test cases would check if custom resource is
+		// getting created as expected and they are handled elsewhere
+		{"replicas field absent", false, 0, false, ""},
+		{"replicas field value less than 0", true, -1, true, "spec.replicas in body should be greater than or equal to 0"},
+	}
+
+	testCtx := framework.NewTestCtx(t)
```

Comment:
This block can be moved to starting as there is not point in declaring the `wmcReplicasFieldValidationTests` struct without getting the namespace

Reply:
Done